src/controller.rs

```diff
@@ -578,32 +598,122 @@ impl AppState {
+                        let order_for_signing_hex = order.clone();
+                        let order_params = order.to_order_params(base_decimals);
+                        (
+                            order_params,
+                            order_for_signing_hex.to_signed_order_hex(
+                                order_params,
+                                current_slot,
+                                sub_account_id,
+                            ),
+                        )
+                    })
+                    .collect();
+
+                for (order, message) in orders_with_hex {
+                    // TODO: generate uuid
+                    let signature = self.wallet.sign_message(message.as_slice())?; // TODO: generate signature
+                    let market_type = match order.market_type {
+                        MarketType::Spot => "spot".to_string(),
+                        MarketType::Perp => "perp".to_string(),
+                    };
+                    let incoming_msg = IncomingSignedMessage {
+                        taker_authority: self.authority().to_string(),
+                        signature: general_purpose::STANDARD.encode(signature),
+                        message: String::from_utf8(message).unwrap(),
+                        signing_authority: self.signer().to_string(),
+                        market_type,
+                        market_index: order.market_index,
+                    };
+
+                    signed_messages.push(incoming_msg);
+                    let hash = digest(signature.to_string().as_bytes());
+                    hashes.push(hash);
+                }
+
+                let client = reqwest::Client::new();
+
+                let swift_orders_url = self
+                    .swift_node
+                    .clone()
+                    .unwrap_or("https://master.swift.drift.trade".to_string())
+                    + "/orders";
+
+                let mut responses: Vec<(String, String)> = Vec::new();
+
+                for msg in signed_messages {
+                    let response = client
+                        .post(&swift_orders_url)
+                        .json(&msg)
+                        .send()
+                        .await
+                        .map_err(|e| {
+                            ControllerError::BadRequest(format!("HTTP request failed: {}", e))
+                        })?;
+
+                    let status = response.status();
+                    let response_text = response.text().await.unwrap_or_default();
+                    responses.push((status.to_string(), response_text));
+                }
 
-        self.send_tx(tx, "place_orders", ctx.ttl).await
+                let signed_msg = SignedMsgResponse {
+                    signed_msg_order_results: hashes
+                        .iter()
+                        .enumerate()
+                        .map(|(i, hash)| SignedMsgOrderResult {
```

Comment:
instead of enumerate and idx into responses can do: `hashes.iter().zip(responses)`